axelrod/strategies/hmm.py

```diff
@@ -0,0 +1,162 @@
+        ------
+        transitions_C and transitions_D are square stochastic matrices:
+            lists of lists with all values in [0, 1] and rows that sum to 1.
+        emission_probabilities is a vector of values in [0, 1]
+        initial_state is an element of range(0, len(emission_probabilities))
+        """
+        self.transitions_C = transitions_C
+        self.transitions_D = transitions_D
+        self.emission_probabilities = emission_probabilities
+        self.state = initial_state
+
+    def is_well_formed(self):
+        if not is_stochastic_matrix(self.transitions_C):
+            return False
+        if not is_stochastic_matrix(self.transitions_D):
+            return False
+        for p in self.emission_probabilities:
+            if (p < 0) or (p > 1):
+                return False
+        if self.state not in range(0, len(self.emission_probabilities)):
+            return False
+        return True
+
+    def move(self, opponent_action):
+        """Changes state and computes the response action."""
+        num_states = len(self.emission_probabilities)
+        if opponent_action == C:
+            next_state = choice(num_states, 1, p=self.transitions_C[self.state])
+        else:
+            next_state = choice(num_states, 1, p=self.transitions_D[self.state])
+        self.state = next_state[0]
+        p = self.emission_probabilities[self.state]
+        action = random_choice(p)
+        return action
+
+
+class HMMPlayer(Player):
+    """Abstract base class for Hidden Markov Model players.
+
+    Names
+
+        - HMMPlayer
+    """
+
+    name = "HMM Player"
+
+    classifier = {
+        'memory_depth': 1,
+        'stochastic': True,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    @init_args
+    def __init__(self, transitions_C=None, transitions_D=None,
+                 emission_probabilities=None, initial_state=0,
+                 initial_action=C):
+        Player.__init__(self)
+        if not transitions_C:
+            transitions_C = [[1]]
+            transitions_D = [[1]]
+            emission_probabilities = [0.5]  # Not stochastic
+            initial_state = 0
+        self.initial_state = initial_state
+        self.initial_action = initial_action
+        self.hmm = SimpleHMM(transitions_C, transitions_D,
+                             emission_probabilities, initial_state)
+        assert self.hmm.is_well_formed()
+        self.classifier['stochastic'] = self.is_stochastic()
+
+    def is_stochastic(self):
+        # If the transitions matrices and emission_probabilities are all 0 or 1
+        # Then the player is stochastic
+        values = set(self.hmm.emission_probabilities)
+        for m in [self.hmm.transitions_C, self.hmm.transitions_D]:
+            for row in m:
+                values.update(row)
+        if not values.issubset({0, 1}):
```

Comment:
`return not values.issubset({0, 1})`.

(This is stylistic, I don't feel strongly about it.)